### ISSUE TITLE
Refresh screen after "Synchronize repository"

### ISSUE THREAD

**Issue by brianbuccola** [zeapo/Android-Password-Store#216]
For a while, I thought that "Synchronize repository" wasn't working properly, or that I somehow misunderstood its function: when I tap it, it looks like it's synchronizing, but afterwards, everything is still the same (for example, I don't see any new password entries).

So instead, I started using "Pull from remote", which works fine.

Only later did I realize that, if I tap "Synchronize repository", then exit the app, then reopen the app, everything is synchronized just fine.

In other words, the synchronization occurs successfully; it's just that the screen doesn't refresh/update automatically. (It does for "Pull from remote", though, as far as I can remember.)

Is this a bug/oversight, or am I misunderstanding how "Synchronize repository" is supposed to work?

**Comment by cqcumber** [comment]
I think so too. If you leave a currently open folder containing passwords and then reenter, the updated passwords appear. So there might really just be missing a refresh at the end of a completed synchronization.
Until I read your post I also thought that synchronization was something different and always used pull and then push. Should be an easy fix, I guess.

**Comment by cqcumber** [comment]
I think in PasswordStore.java:506 onActivityResult there is no case for REQUEST_SYNC so no updateListAdapter() is called.

**Comment by zeapo** [comment]
Nice catch!

**Comment by zeapo** [comment]
fixed in 4663ea371edd00068be2cb423320044692b3e965
